_posts/2019-08-05-image-itk.md

```diff
@@ -0,0 +1,686 @@
+  <line x1="27" y1="0" x2="27" y2="33" style="stroke-width:2" />
+
+  <!-- Colored Rectangle -->
+  <polygon points="0.000000,0.000000 27.530335,0.000000 27.530335,33.941765 0.000000,33.941765" style="fill:#ECB172A0;stroke-width:0"/>
+
+  <!-- Text -->
+  <text x="13.765167" y="53.941765" font-size="1.0rem" font-weight="100" text-anchor="middle" >1</text>
+  <text x="47.530335" y="16.970882" font-size="1.0rem" font-weight="100" text-anchor="middle" transform="rotate(0,47.530335,16.970882)">3</text>
+
+
+  <!-- Horizontal lines -->
+  <line x1="97" y1="0" x2="173" y2="76" style="stroke-width:2" />
+  <line x1="97" y1="82" x2="173" y2="158" style="stroke-width:2" />
+
+  <!-- Vertical lines -->
+  <line x1="97" y1="0" x2="97" y2="82" style="stroke-width:2" />
+  <line x1="173" y1="76" x2="173" y2="158" style="stroke-width:2" />
+
+  <!-- Colored Rectangle -->
+  <polygon points="97.000000,0.000000 173.470588,76.470588 173.470588,158.846826 97.000000,82.376238" style="fill:#ECB172A0;stroke-width:0"/>
+
+  <!-- Horizontal lines -->
+  <line x1="97" y1="0" x2="179" y2="0" style="stroke-width:2" />
+  <line x1="173" y1="76" x2="255" y2="76" style="stroke-width:2" />
+
+  <!-- Vertical lines -->
+  <line x1="97" y1="0" x2="173" y2="76" style="stroke-width:2" />
+  <line x1="179" y1="0" x2="255" y2="76" style="stroke-width:2" />
+
+  <!-- Colored Rectangle -->
+  <polygon points="97.000000,0.000000 179.376238,0.000000 255.846826,76.470588 173.470588,76.470588" style="fill:#ECB172A0;stroke-width:0"/>
+
+  <!-- Horizontal lines -->
+  <line x1="173" y1="76" x2="255" y2="76" style="stroke-width:2" />
+  <line x1="173" y1="158" x2="255" y2="158" style="stroke-width:2" />
+
+  <!-- Vertical lines -->
+  <line x1="173" y1="76" x2="173" y2="158" style="stroke-width:2" />
+  <line x1="255" y1="76" x2="255" y2="158" style="stroke-width:2" />
+
+  <!-- Colored Rectangle -->
+  <polygon points="173.470588,76.470588 255.846826,76.470588 255.846826,158.846826 173.470588,158.846826" style="fill:#ECB172A0;stroke-width:0"/>
+
+  <!-- Text -->
+  <text x="214.658707" y="178.846826" font-size="1.0rem" font-weight="100" text-anchor="middle" >64</text>
+  <text x="275.846826" y="117.658707" font-size="1.0rem" font-weight="100" text-anchor="middle" transform="rotate(0,275.846826,117.658707)">64</text>
+  <text x="125.235294" y="140.611532" font-size="1.0rem" font-weight="100" text-anchor="middle" transform="rotate(45,125.235294,140.611532)">101</text>
+</svg>
+</td>
+</tr>
+</table>
+
+
+```python
+# Convert data to float32 for computation¶
+import numpy as np
+imgs = imgs.astype(np.float32)
+psf = psf.astype(np.float32)
+
+# Apply Richardson-Lucy Deconvolution¶
+def richardson_lucy_deconvolution(img, psf, iterations=1):
+    """ Apply deconvolution to a single chunk of data """
+    import itk
+
+    img = img.squeeze()  # remove leading two length-one dimensions
+    psf = psf.squeeze()  # remove leading two length-one dimensions
```

Comment:
Just as a note, would personally avoid `.squeeze()`. Have seen it result in buggier code in practice.

Reply:
Done.  Thanks.